### ISSUE TITLE
default_hp_metric should be False or better explained in the docs

### ISSUE THREAD

**Issue by jadolfbr** [PyTorchLightning/pytorch-lightning#7764]
## 🚀 Feature
Add more docs to `default_hp_metric` in `TensorBoardLogger`.  If this is used as default, all logging of hyper parameters with data is completely ignored - such as `self.logger.log_hyperparameters()` or `self.logger.experiment.log_hparams()`

Also, without this, the only way to even see any hyperparameters in the hparams tab is to use self.save_hyperparameters. 

This is not very clearly explained in the `TensorBoardLogger` docs, but is in a note in the Logging page. Would be nice to have better documentation of the requirements - or even fail if the user tries to do this instead of getting a blank tab.  Ideally, I would say that this should not be the default or somehow enable both to occur.  Not being able to call log_hparams directly with the writer without another option is non-ideal, especially when converting to lightning from general pytorch. 

### Motivation

Lack of documentation, odd behavior for a general, broadly-used tensor board feature, no clear failure, and no clear way to proceed through code errors. 

### Pitch

1) Better documentation in the TensorBoardLogger. 
2) clear failure when trying to log_hyperparameters
3) Ideally, change the default behavior to be what is expected for tensorboard users. 

### Alternatives

4) Use the default hp metric at the end of train/test if nothing else is given.

**Comment by justusschock** [comment]
Hey @jadolfbr 

Thanks for the issue, you're right!

Would you mind opening a PR with your suggested changes for the docs)?

**Comment by jadolfbr** [comment]
Will do!  I’m not familiar with the doc organization.  Where should I make the changes for a PR? Do you do a fork model for PRs?  Thanks!

**Comment by justusschock** [comment]
The easiest could be to add that to the respecitve logger docstrings. Otherwise you should find everything at `/docs/source/common/loggers.rst `

Yes, you have to fork the repo to create a PR :)

**Comment by kaushikb11** [comment]
Hi @jadolfbr, you could refer to these Contributing [guidelines](https://pytorch-lightning.readthedocs.io/en/latest/generated/CONTRIBUTING.html#guidelines) for submitting a PR!

**Comment by jadolfbr** [comment]
Thanks guys!
